src/stacks-hierarchy/Root_TabNavigatorStack/TabNav_TicketingStack/Ticketing_TicketTabNavStack/TicketTabNav_PurchaseTabScreen/Components/RecentFareContracts/RecentFareContractComponent.tsx

```diff
@@ -74,26 +85,55 @@ export const RecentFareContractComponent = ({
         RecentFareContractsTexts.repeatPurchase.label,
       )} ${modeInfo} ${travellerInfo}`;
     }
-
-    const zoneInfo = `${
+    const zoneInfo =
       fromZoneName === toZoneName
         ? `${t(
             RecentFareContractsTexts.a11yPreLabels.zones.oneZone,
           )} ${fromZoneName}`
         : `${t(
             RecentFareContractsTexts.a11yPreLabels.zones.multipleZones,
-          )} ${fromZoneName}, ${toZoneName}`
-    }`;
+          )} ${fromZoneName}, ${toZoneName}`;
+
+    const harborInfo = () => {
+      if (pointToPointValidity?.fromPlace && pointToPointValidity?.toPlace) {
+        const fromName =
+          harborsQuery.data?.find(
+            (sp) => sp.id === pointToPointValidity.fromPlace,
+          )?.name ?? '';
+        const toName =
+          harborsQuery.data?.find(
+            (sp) => sp.id === pointToPointValidity.toPlace,
+          )?.name ?? '';
+        return direction === TravelRightDirection.Both
+          ? t(
+              RecentFareContractsTexts.a11yPreLabels.harbors.returnTrip(
+                fromName,
+                toName,
+              ),
+            )
+          : t(
+              RecentFareContractsTexts.a11yPreLabels.harbors.oneWayTrip(
+                fromName,
+                toName,
+              ),
+            );
+      }
+      return '';
+    };
```

Comment:
In recent times I have become less of a fan of creating these functions during the component render, since that means it is recreated on every render. Creating functions during render (without `useCallback`) can lead to errors if used incorrectly, like in a `useEffect`.

I think it is better to either:
- Take it out of the component render as it's own function someplace else in the file.
- Just calculate the `harborInfo` during render instead of a reactive function for it.
- If the calculation of `harborInfo` is heavy and might give performance issues when done on every render, then wrap it with `useMemo`.

And after writing this and reading the code one more time I see that the `harborInfo` function is actually inside the `returnAccessibilityLabel` function. Then my comments on `harborInfo` doesn't really apply, but I still leave them here as they still apply for the `returnAccessibilityLabel` function. Not something you need to fix now.